kodi_addon_checker/check_addon.py

```diff
@@ -351,3 +356,47 @@ def _logWarning(error_counter, warning_string):
     colorPrint("WARNING: %s" % warning_string, "35")
     error_counter["warnings"] = error_counter["warnings"] + 1
     return error_counter
+
+
+def get_addons(xml_url):
+    """
+        addon.xml for the target Kodi version
+    """
+    content = requests.get(xml_url).content
+    tree = ET.fromstring(content)
+
+    return {
+        a.get("id"): a.get("version")
+        for a in tree.findall("addon")
+    }
+
+
+def get_users_dependencies(addon_path):
+    """
+        User's addon.xml from pull request
+    """
+    addon_xml_path = os.path.join(addon_path, "addon.xml")
+
+    tree = ET.parse(addon_xml_path).getroot()
+
+    return {
+        i.get("addon"): i.get("version")
+        for i in tree.findall("requires/import")
+    }
+
+
+def _check_dependencies(error_counter, addon_path, repo_addons):
+    deps = get_users_dependencies(addon_path)
+    ignore = ['xbmc.json', 'xbmc.gui', 'xbmc.json', 'xbmc.metadata', 'xbmc.python']
+
+    for required_addon, required_version in deps.items():
+        if (required_addon not in repo_addons) and (required_addon not in ignore):
+            error_counter = _logProblem(error_counter, "Unrecognized requirement: %s " % required_addon)
```

Comment:
I think we can word this better, how about `Required addon "%s" not available in current repository.`